### ISSUE TITLE
Wrap filenames in Explorer

### ISSUE THREAD

**Issue by sanmai-NL** [microsoft/vscode#111411]
Currently, the Explorer has to be sized to accomodate for the file with the longest name to be able to view all filenames at a glance. This wastes a lot of screen space when the range of filename lengths is wide. 

I propose to let filenames wrap instead of overflowing with an ellipsis placeholder.

**Comment by isidorn** [comment]
No plans to add this. But let's keep as a feature request for now.

**Comment by sanmai-NL** [comment]
@isidorn What's the ’confused’ smiley about?

**Comment by isidorn** [comment]
I did not put it.

**Comment by sanmai-NL** [comment]
Ah .. That's right. I hope the request is clear for you then?

**Comment by IllusionMH** [comment]
Confused smile is from me (should be shown when hovered over smile):
1. Have you tried to enable `"workbench.list.horizontalScrolling": true` to see scroll instead of ellipsis? Should be easy to scroll to the side to see it full (if overflow mostly caused by shift because of nesting)
1. AFAIK currently explorer uses virtualized tree wit fixed height and if it would be changed to dynamic might negatively impact performance (however I hope that calculations can be only if flag enabled and won't affect performance for other, or that I'm just wrong in my assumption about performance implications)
1. How it should work for (deeply) nested folders that also overflow current width? If folders take 2-3 lines this will make visual scanning (for nesting) only harder. BUT I guess this will be under the option so my concerns here can be skipped, since I won't enable it.

**Comment by sanmai-NL** [comment]
I know, I just confused your handles.

1. That's not a solution to my problem. I can just as easily (or more) make the Explorer larger and shrink it back. The problem is, the full filenames are not shown *at a glance* when some are longer than usual.
2. Don't know. I can't image this being a problem. By the way, filenames can be multi-line anyway, though VS Code truncates the part after the first newline (I just tested).
3. Sure, and the feature request is about clipped filenames, the final path component, not full paths.

**Comment by IllusionMH** [comment]
Sorry, I shouldn't point obvious thing about reactions even if I didn't expect that handles can be confused.

1. Resizing will move/resize editors, scroll will help to avoid that therefore I proposed it. As personal experience suggests - there are people who would like avoid shifting editors, but scroll will be OK and they just don't know that it's possible to enable it. Now as you stated - doesn't work your case.

1. My main concern: If I understand correctly currently all explorer "rows" have fixed height.
https://github.com/microsoft/vscode/blob/8ee75c19bb39c7433a042647d29d34327a3d55f6/src/vs/workbench/contrib/files/browser/views/explorerViewer.ts#L65
If any item in tree can trigger wrap for itself that will require recalculations + will make scroll(bar) inaccurate, or will require recalculations for all items (which obviously taxing) to have proper scroll(bar) behavior. And as usual recalculations on resizes.
Not a concern if can be implemented behind the option so wont affect others.
Personally it would be interesting for me to see how it might be solved in VS Code, because virtual lists on web pages sometimes have bad performance for items with different height.

1. A bit strange to have wrapping only on files, but not on folders that overflow if (I understand your comment correctly). 

---
To sum up:
I'm not against the idea, but a bit worried about possible impact on Explorer performance and inconsistency between wrapping behavior for folders & files in proposal.

**Comment by sanmai-NL** [comment]
Thanks for sharing your analysis. (I confused you reaction mainly because of its timing, only partly because of your handle, I think.)

Can't the number of rows be calculated once based on filename length spread? Wrapping doesn't mean it should occur by recalculation upon overflow.

I see the setting you referred also warns of negative performance impact. I'm used to developing other kinds of software where this kind of thing would't be expected to be a performance drag.

**Comment by khaosdoctor** [comment]
Wrapping the lines or doing some smart ellipsis would help a lot, for now the horizontal scroll solves my problem, but being able to see it all with wrapped filenames would be awesome.

Other idea is to add an option to enable the full file name being shown on hover.

**Comment by zioth** [comment]
This feature request has my vote too. My company uses extremely long filenames, so wrapping them would be helpful.

**Comment by scotthugh** [comment]
why wrap them, which is not possible as of todays date, why not just show the entire filename on hover like webstorm?

**Comment by scotthugh** [comment]
Have you seen how webstorm handles this?

When hovering over a file name which is hidden under the overflow is becomes visible on top of the editor. And it's instant. Not like the tooltip we have now that takes 1-2 seconds to appear and is far too difficult to decode at a glance.

**Comment by starball5** [comment]
Related on Stack Overflow: [How can I toggle the Word Wrap on the Left sidebar of Visual Studio Code?](https://stackoverflow.com/q/78565573/11107541)

**Comment by buedaswag** [comment]
This feature has been requested independently in several other issues:                                                                                                               
- #197785
- #214269
- #233509                                                                                                                                                                                                                
                                                                                                                                                                                                                           Would be great to see this considered for implementation. 
The workaround of widening the sidebar wastes screen real estate, especially on laptops.
